osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/processors/GetLightSensorStatusRequestMessageProcessor.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.iec60870.infra.messaging.processors;
+
+import java.io.IOException;
+
+import org.openmuc.j60870.ASdu;
+import org.openmuc.j60870.ASduType;
+import org.openmuc.j60870.CauseOfTransmission;
+import org.openmuc.j60870.ie.IeQualifierOfInterrogation;
+import org.openmuc.j60870.ie.InformationObject;
+import org.opensmartgridplatform.adapter.protocol.iec60870.domain.services.ClientConnection;
+import org.opensmartgridplatform.adapter.protocol.iec60870.domain.valueobjects.LogItem;
+import org.opensmartgridplatform.adapter.protocol.iec60870.domain.valueobjects.RequestMetadata;
+import org.opensmartgridplatform.adapter.protocol.iec60870.infra.messaging.AbstractMessageProcessor;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.ProtocolAdapterException;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+/**
+ * Class for processing get light sensor status requests.
+ */
+@Component
+public class GetLightSensorStatusRequestMessageProcessor extends AbstractMessageProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetLightSensorStatusRequestMessageProcessor.class);
+
+    public GetLightSensorStatusRequestMessageProcessor() {
+        super(MessageType.GET_LIGHT_SENSOR_STATUS);
+    }
+
+    @Override
+    public void process(final ClientConnection deviceConnection, final RequestMetadata requestMetadata)
+            throws ProtocolAdapterException {
+
+        final String deviceIdentification = requestMetadata.getDeviceIdentification();
+        final String organisationIdentification = requestMetadata.getOrganisationIdentification();
+
+        LOGGER.info("Get light sensor status request for IEC60870 device {} for organisation {}", deviceIdentification,
+                organisationIdentification);
+
+        try {
+            // Perform general interrogation
+            final int ieQualifierOfInterrogationValue = 20;
+            final int originatorAddress = 0;
+            final int commonAddress = deviceConnection.getConnectionParameters().getCommonAddress();
+
+            deviceConnection.getConnection()
+                    .interrogation(commonAddress, CauseOfTransmission.ACTIVATION,
+                            new IeQualifierOfInterrogation(ieQualifierOfInterrogationValue));
```

Comment:
The get health status message processor could probably be removed once the connection restore mechanism uses the connect requests. 
The get light sensor status should unfortunately use the general interrogation as well, because the read command will not be supported in the actual light measurement device implementation.
Refactored duplicated code into a new GeneralInterrogationService class.